pkgs/applications/version-management/forgejo/default.nix

```
@@ -1,25 +1,72 @@
-{ fetchurl, gitea, lib }:
+{ lib
+, stdenv
+, buildGoPackage
+, fetchurl
+, makeWrapper
+, git
+, bash
+, gzip
+, openssh
+, pam
+, pamSupport ? true
+, sqliteSupport ? true
+}:
 
-gitea.overrideAttrs (old: rec {
+let
+  tags = lib.optional pamSupport "pam"
+    ++ lib.optional sqliteSupport "sqlite sqlite_unlock_notify";
+  tagsString = lib.concatStringsSep " " tags;
+in
+
+buildGoPackage rec {
   pname = "forgejo";
-  version = "1.18.0-rc1-1";
+  version = "1.18.0-1";
 
   src = fetchurl {
     name = "${pname}-src-${version}.tar.gz";
     # see https://codeberg.org/forgejo/forgejo/releases
-    url = "https://codeberg.org/attachments/976c426a-3e04-49ff-9762-47fab50624a3";
-    hash = "sha256-kreBMHlMVB1UeG67zMbszGrgjaROateCRswH7GrKnEw=";
+    url = "https://codeberg.org/attachments/c829784c-3b85-4996-9dc6-09e12e40a93a";
+    hash = "sha256-42bR18T5ATVyhPejeHpOTkeP6VwYu5HFfei8oMjRJy8=";
   };
 
-  postInstall = old.postInstall or "" + ''
-    mv $out/bin/{${old.pname},${pname}}
+  outputs = [ "out" "data" ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  buildInputs = lib.optional pamSupport pam;
+
+  patches = [
+    ./../gitea/static-root-path.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace modules/setting/setting.go --subst-var data
   '';
 
+  preBuild = ''
+    export buildFlagsArray=(
+      -tags="${tagsString}"
+      -ldflags='-X "main.Version=${version}" -X "main.Tags=${tagsString}"'
+    )
+  '';
```

```suggestion
  tags = lib.optional pamSupport "pam"
    ++ lib.optional sqliteSupport "sqlite sqlite_unlock_notify";
  
  ldflags = [
    "-X main.Version=${version}"
    "-X main.Tags=${lib.concatStringsSep " " tags}"
  ];
```

---

Error build:
```
Building subPackage code.gitea.io/gitea
cannot find package "sqlite_unlock_notify" in any of:
        /nix/store/x4p5yg47ai209jin1phhj4fc2iw3b5ix-go-1.19.4/share/go/src/sqlite_unlock_notify (from $GOROOT)
        /build/go/src/sqlite_unlock_notify (from $GOPATH)
cannot find package "-ldflags=-X main.Version=1.18.2-0 -X main.Tags=pam sqlite sqlite_unlock_notify" in any of:
        /nix/store/x4p5yg47ai209jin1phhj4fc2iw3b5ix-go-1.19.4/share/go/src/-ldflags=-X main.Version=1.18.2-0 -X main.Tags=pam sqlite sqlite_unlock_notify (from $GOROOT)
        /build/go/src/-ldflags=-X main.Version=1.18.2-0 -X main.Tags=pam sqlite sqlite_unlock_notify (from $GOPATH)
cannot find package "-p" in any of:
        /nix/store/x4p5yg47ai209jin1phhj4fc2iw3b5ix-go-1.19.4/share/go/src/-p (from $GOROOT)
        /build/go/src/-p (from $GOPATH)
cannot find package "32" in any of:
        /nix/store/x4p5yg47ai209jin1phhj4fc2iw3b5ix-go-1.19.4/share/go/src/32 (from $GOROOT)
        /build/go/src/32 (from $GOPATH)

```

---

Use
```suggestion
  tags = lib.optional pamSupport "pam"
    ++ lib.optionals sqliteSupport [ "sqlite" "sqlite_unlock_notify" ];

  ldflags = [
    "-X main.Version=${version}"
    "-X 'main.Tags=${lib.concatStringsSep " " tags}'"
  ];
```
instead.
Note the `lib.optionals` and single quotes `'` in `"-X 'main.Tags=${lib.concatStringsSep " " tags}'"`

```
❯ nix-build -A forgejo
/nix/store/k818d0m46hc380cclhdrp0sfkyi414hc-forgejo-1.18.2-0
❯ ./result/bin/gitea --version
Gitea version 1.18.2-0 built with go1.19.4 : pam, sqlite, sqlite_unlock_notify
```

---

Thanks!